src/discussions/common/Confirmation.jsx

```diff
@@ -26,25 +27,32 @@ function DeleteConfirmation({
       </ModalDialog.Body>
       <ModalDialog.Footer>
         <ActionRow>
-          <ModalDialog.CloseButton variant="tertiary">
+          <ModalDialog.CloseButton variant={onDelete ? 'tertiary' : 'default'}>
             {intl.formatMessage(messages.deleteConfirmationCancel)}
           </ModalDialog.CloseButton>
-          <Button variant="primary" onClick={onDelete}>
-            {intl.formatMessage(messages.deleteConfirmationDelete)}
+          <Button variant={onDelete ? 'primary' : 'danger'} onClick={onDelete || onReport}>
+            {onDelete
+              ? intl.formatMessage(messages.deleteConfirmationDelete)
+              : intl.formatMessage(messages.reportConfirmationConfirm) }
           </Button>
         </ActionRow>
       </ModalDialog.Footer>
     </ModalDialog>
   );
 }
 
-DeleteConfirmation.propTypes = {
+Confirmation.propTypes = {
   intl: intlShape.isRequired,
   isOpen: PropTypes.bool.isRequired,
   onClose: PropTypes.func.isRequired,
-  onDelete: PropTypes.func.isRequired,
+  onDelete: PropTypes.func,
+  onReport: PropTypes.func,
```

Comment:
instead of adding two different props deleteConfirm and reportConfirm rename it to comfirmAction and make is require

Reply:
changed